sklearn/utils/_repr_html/tests/test_js.py

```diff
@@ -0,0 +1,134 @@
+import socket
+import threading
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from pathlib import Path
+
+import pytest
+
+
+@pytest.fixture(scope="session", autouse=True)
+def check_playwright():
+    """Skip tests if playwright is not installed.
+
+    This fixture is used by the next fixture (which is autouse) to skip all tests
+    if playwright is not installed."""
+    return pytest.importorskip("playwright")
+
+
+@pytest.fixture
+def local_server(request):
+    """Start a simple HTTP server that serves custom HTML per test.
+
+    Usage :
+
+    ```python
+    def test_something(page, local_server):
+        url, set_html_response = local_server
+        set_html_response("<html>...</html>")
+        page.goto(local_server)
+        ...
+    ```
+    """
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("127.0.0.1", 0))
+        PORT = s.getsockname()[1]
+
+    html_content = "<html><body>Default</body></html>"
+
+    def set_html_response(content):
+        nonlocal html_content
+        html_content = content
+
+    class Handler(BaseHTTPRequestHandler):
+        def do_GET(self):
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+            self.wfile.write(html_content.encode("utf-8"))
+
+        # suppress logging
+        def log_message(self, format, *args):
+            return
+
+    httpd = HTTPServer(("127.0.0.1", PORT), Handler)
+    thread = threading.Thread(target=httpd.serve_forever, daemon=True)
+    thread.start()
+
+    yield f"http://127.0.0.1:{PORT}", set_html_response
+
+    httpd.shutdown()
+
+
+def _make_page(body):
+    """Helper to create a HTML page that includes `estimator.js` and the given body."""
+
+    js_path = Path(__file__).parent.parent / "estimator.js"
+    with open(js_path, "r", encoding="utf-8") as f:
+        script = f.read()
+
+    return f"""
+    <html>
+      <head>
+      <script>{script}</script>
+      </head>
+      <body>
+        {body}
+      </body>
+    </html>
+    """
+
+
+def test_copy_paste(page, local_server):
+    """Test that copyToClipboard copies the right text to the clipboard.
+
+    Test requires clipboard permissions, which are granted through page's context.
+    Assertion is done by reading back the clipboard content from the browser.
+    This is easier than writing a cross platform clipboard reader.
+    """
+    url, set_html_response = local_server
+
+    copy_paste_html = _make_page(
+        '<div class="sk-toggleable__content" data-param-prefix="prefix"/>'
+    )
+
+    set_html_response(copy_paste_html)
+    page.context.grant_permissions(["clipboard-read", "clipboard-write"])
+    page.goto(url)
+    page.evaluate(
+        "copyToClipboard('test', document.querySelector('.sk-toggleable__content'))"
+    )
+    clipboard_content = page.evaluate("navigator.clipboard.readText()")
+    assert clipboard_content == "prefixtest"
```

Comment:
Can we add a comment here to explain why the expected content is `"prefixtest"`. From reading the code I thought it would be `"prefix"` (the value of the `data-param-prefix` attribute).

That intuition is wrong, and I can't work out why the assert is what it is. So maybe other people from the future will also struggle. Hence the idea with adding a comment.